### DISCUSSION TITLE
errors with JSON_ASSERT

### DISCUSSION THREAD

**Discussion by SpookyWookyMan** [nlohmann/json#2265] [Q&A] [answered]
Hello, today I found this wonderful library, but when I tried to include the .hpp file on my project I got the following error:
E0020 identifier "assert" is undefined

I found this extremely odd since only 9 of 131 uses of the macro JSON_ASSERT gave me the error:
Line 14001
Line 14078
Line 14097
Line 14401
Line 14480
Line 14536
Line 14762
Line 14861
Line 14914

I don't really know how to fix this so if you could give me a hand it'd be greatly appreciated!

**Comment by nlohmann** [comment, marked as answer]
In order to asses your issue, we need the following information:

- Please describe the steps to reproduce the issue. Can you provide a small but working code example?

- What is the expected behavior?

- And what is the actual behavior instead?

- Which compiler and operating system are you using? Is it a [supported compiler](https://github.com/nlohmann/json#supported-compilers)?

- If you experience a compilation error: can you [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests)?

**Reply by SpookyWookyMan** [reply]
I literally just inserted the single include in my project and it gave me the error, havent tried using any features from the library to see if it works , but still, I find it odd.
Windows 10 Pro 64bit
Microsoft Visual C++ 2019 / Build Tools 16.3.1+1def00d3d (and possibly later)
![image](https://user-images.githubusercontent.com/48411770/87386201-56206d80-c576-11ea-803e-fcccbec19369.png)

**Reply by nlohmann** [reply]
That is really strange as the CI successfully compiles and tests the library with MSVC 2019.

- Can you compile and run the unit tests?

**Reply by SpookyWookyMan** [reply]
Will do as soon as I get ahold of my pc

**Reply by SpookyWookyMan** [reply]
Ok, so I booted up visual studio and now the error is gone for some reason lol
Thank you for replying anyways
